system/lib/fetch/wasmfs.cpp

```diff
@@ -0,0 +1,55 @@
+// // Copyright 2021 The Emscripten Authors.  All rights reserved.
+// // Emscripten is available under two separate licenses, the MIT license and the
+// // University of Illinois/NCSA Open Source License.  Both these licenses can be
+// // found in the LICENSE file.
+
+#include <stdlib.h>
+#define __NEED_struct_iovec
+#include <emscripten/emscripten.h>
+#include <wasi/api.h>
+
+extern "C" {
```

Comment:
For C++ style, perhaps we can just use the clang-format settings for say LLVM or Binaryen? We can also leave that for later, but meanwhile, I'd suggest we use something more like the existing code we have, like `system/lib/embind/bind.cpp` (in particular: no indentation inside extern C)

Reply:
Previously projects could just be compiled in C. Are there likely to be any implications of now adding in a C++ library?

Reply:
IIUC, @curiousdannii exported functions need to be C functions (to avoid C++ name mangling). If this is what you're asking.

Reply:
@kripken I used clang-format for now for consistency

Reply:
The main idea is that we don't indent stuff that lives inside `namespace {` or inside `extern "C" {`

Reply:
Got it. I think when I ran `clang-format` locally it removed the indentations.